### PR TITLE
Feat/index reusable content

### PR DESCRIPTION
**Pull request by bkapustik**
# Motivation

* Adds support for mapping reusable content types to Lucene documents.
* Solves the following issue #62

### CODE REVIEW

File: examples/DancingGoat/Search/ReusableContentItemsIndexingStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    )
+    {
+        this.urlRetriever = urlRetriever;
+        this.webPageMapper = webPageMapper;
+        this.queryExecutor = queryExecutor;
+        this.htmlSanitizer = htmlSanitizer;
+        this.webCrawler = webCrawler;
+    }
+
+    public override async Task<Document?> MapToLuceneDocumentOrNull(IIndexEventItemModel item)
```
Comment (michalJakubis) [inline examples/DancingGoat/Search/ReusableContentItemsIndexingStrategy.cs:44]: Describe the function of this method with doc comment.

File: src/Kentico.Xperience.Lucene.Core/InfoModels/LuceneReusableContentTypeItem/ILuceneReusableContentTypeItemInfoProvider.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+using CMS.DataEngine;
+
+namespace Kentico.Xperience.Lucene.Core;
+
+/// <summary>
+/// Declares members for <see cref="LuceneReusableContentTypeItemInfo"/> management.
+/// </summary>
+public partial interface ILuceneReusableContentTypeItemInfoProvider
+{
+    void BulkDelete(IWhereCondition where, BulkDeleteSettings? settings = null);
```
Comment (michalJakubis) [inline src/Kentico.Xperience.Lucene.Core/InfoModels/LuceneReusableContentTypeItem/ILuceneReusableContentTypeItemInfoProvider.cs:10]: Describe also this API with a comment

File: src/Kentico.Xperience.Lucene.Core/InfoModels/LuceneReusableContentTypeItem/LuceneReusableContentTypeItemInfoProvider.generated.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+using CMS.DataEngine;
+
+namespace Kentico.Xperience.Lucene.Core;
+
+/// <summary>
+/// Class providing <see cref="LuceneReusableContentTypeItemInfo"/> management.
+/// </summary>
+[ProviderInterface(typeof(ILuceneReusableContentTypeItemInfoProvider))]
+public partial class LuceneReusableContentTypeItemInfoProvider : AbstractInfoProvider<LuceneReusableContentTypeItemInfo, LuceneReusableContentTypeItemInfoProvider>, ILuceneReusableContentTypeItemInfoProvider
```
Comment (michalJakubis) [inline src/Kentico.Xperience.Lucene.Core/InfoModels/LuceneReusableContentTypeItem/LuceneReusableContentTypeItemInfoProvider.generated.cs:9]: There is not need for dedicated InfoProvider. Just refactor Info class with IInfoProvider approach and remove this LuceneReusableContentTypeItemInfoProvider.

**Review by michalJakubis — COMMENTED** [review]
Looks good to me. Waiting for TW review.

**Review by michalJakubis — APPROVED** [review]

**Review by laurao123 — APPROVED** [review]
The documentation update looks good :)

**Pull request merged**